WordPress/gutenberg
# Output post classes in the editor

## What?

This pull request addresses issue [ #33887](https://github.com/WordPress/gutenberg/issues/33887) (and I think [#10640](https://github.com/WordPress/gutenberg/issues/10640) is related to this) by introducing post classes within the query loop in the Gutenberg editor.

Note:
I would like to know if the approach in this pull request is in the right redirection.

## Why?

This allows users to target specific elements and apply custom styles to them.

## How?

Extends the WordPress REST API to include post classes in the response data for all public post types.

### Alternatives

- Alternative using a new REST API route to retrieve CSS classes of a specific post: [#60645](https://github.com/WordPress/gutenberg/pull/60645)

- Recreating all the classes in JS along with a JS filter could be an alternative solution.

## Testing Instructions




Insert a Query block and observe the post classes added next to the `wp-post-block` class in the block inspector.

### Testing Instructions for Keyboard

--

## Screenshots or screencast 
![image](https://github.com/WordPress/gutenberg/assets/50170696/7f42009f-ae4b-4ddb-91d7-58d79c82098a)
